pyccel/parser/semantic.py

```diff
@@ -2967,6 +2967,26 @@ def _visit_FunctionDef(self, expr, **settings):
         if len(not_used) >= 1:
             errors.report(UNUSED_DECORATORS, symbol=', '.join(not_used), severity='warning')
 
+        # Check if an argument is modified
+        
+        def check_if_const(arguments, code_block):
+            lines = code_block.body
+            for line in lines:
+                if isinstance(line, For) :
+                    check_if_const(arguments, line.body)
```

Comment:
You should be able to use [`arg.get_user_nodes(Assign)`](https://github.com/pyccel/pyccel/blob/master/pyccel/ast/basic.py#L92-L107) for this instead of recursing. This limits the chance of you missing cases, eg:
- if
- while